flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -263,6 +298,36 @@ private static LocalResource registerLocalResource(FileSystem fs, Path remoteRsr
 		return localResource;
 	}
 
+	/**
+	 * Register a local resource with resource info. The resource info may contains multiple parts.
+	 * For example, RemotePath;[resourceSize;resourceModificationTime;LocalResourceVisibility]
+	 * @param resourceInfoStr resource info string
+	 * @param yarnConfig yarn configuration
+	 * @return local resource tuple, f0 is filename, f1 is local resource.
+	 */
+	private static Tuple2<String, LocalResource> registerLocalResource(
```

Comment:
can also split into `registerLocalResource` and `registerPreloadedResource` - or if we think this is not reused somewhere else, we can always merge it with the `setUp...` functions

Reply:
I prefer the name `registerLocalResource`. It has different overloadings.

```
private static LocalResource registerLocalResource(Path remoteRsrcPath, long resourceSize, long resourceModificationTime, LocalResourceVisibility visibility)

private static LocalResource registerLocalResource(FileSystem fs, Path remoteRsrcPath)

private static Tuple2<String, LocalResource> registerLocalResource(String resourceDesStr, Configuration yarnConfig)
```

Reply:
This method is quite tricky for introducing an in-place pattern "RemotePath;[resourceSize;resourceModificationTime;LocalResourceVisibility]" to parse configuration...I think it works as is but +0 for this patch. @walterddr you can merge this patch if you think it is good to go.

Reply:
@TisonKun: sorry could you elaborate the "in-place" pattern you mentioned? and are you a +0 on the current PR implementation of on the split to `registerLocal...` & `registerPreloaded...` ?

Reply:
@TisonKun correct me if I were wrong. I think you are also having concerns regarding the branch out paradigm here

Reiterating my previous comment: since the code path to deal with preuploaded resource is different from the local resource (given the if-else pattern) we can probably split it into different methods? what do you think @TisonKun @wangyang0918